### DISCUSSION TITLE
astro/cli "asto add" idea

### DISCUSSION THREAD

**Discussion by dayvster** [withastro/roadmap#396] [Proposal]
Currently if we wish to add an integration we have `astro add`, however removing or toggling integrations is done manually.

It would be cool if we could do all of that via the CLI

**Proposed solution**:
- `astro remove 'integration name'` removes the integration from package.json and any and all config files associated with it get ammended
- `astro toggle 'integration name'` amends the config files but keeps the integration present in package.json
- `astro enable 'integration name'` same as toggle but only enables the integration
- `astro disable 'integration name'` same as toggle but only disabled the integration

**Comment by neelsudhakaran** [comment]
This sounds really cool! Great idea 👍

**Comment by neelsudhakaran** [comment]
@dayvster—are you interested in opening a formal RFC for this or may I? Totally up to you, and we can also wait to garner more interest on this first!

**Comment by dayvster** [comment]
I'd love to work on this feature myself, thanks for the offer though. Glad you like it.

If I fail to secure enough time in my schedule I'll let you know ;)

**Comment by riywo** [comment]
FYI: I submitted another idea which could cover this feature: https://github.com/withastro/rfcs/discussions/428

**Comment by florian-lefebvre** [comment]
The problem is `astro add` do not only make changes to the astro config but can add any files. This is simply a starting point and then the user can customize so it's not revertable that easily. Anyways, it's probably better to gather interest in one place, see https://github.com/withastro/roadmap/discussions/522
